### DISCUSSION TITLE
Problem when fetching data async in preprocessor - what to do?

### DISCUSSION THREAD

**Discussion by NicolajHansenPersonal** [dotnet/docfx#9865] [General]
I have a task that has me calling an api to fetch some data, that I would like to be rendered in one of my templates. Setting up the preprocessor .js file, using the exports.transform function to alter the model, I run into some problems when trying to use "await fetch()" to get the data. It simply wont work.
If I place the call in a try catch, the error message I get is "fetch is not defined"?

Anyone with some good ideas how to work with API calls in docfx and be able to work with the data before using it :)

Kind regards
Nicolaj

**Comment by filzrev** [comment]
There are similar questions.
- https://github.com/dotnet/docfx/issues/4226
- `https://github.com/sebastienros/jint/issues/442`
- https://stackoverflow.com/questions/55211689/simple-api-call-within-the-context-of-docfx-preprocessor

As far as I knows docfx don't providing Jint customization options.
So available option is to create custom docfx plugin.

**Reply by NicolajHansenPersonal** [reply]
thank you :)
